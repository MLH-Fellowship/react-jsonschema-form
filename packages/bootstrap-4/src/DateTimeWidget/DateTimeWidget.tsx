import React from "react";
import { utils } from "@rjsf/core";
import TextWidget, { TextWidgetProps } from "../TextWidget";

const { localToUTC, utcToLocal } = utils;

const DateTimeWidget = (props: TextWidgetProps) => {
  const value = utcToLocal(props.value);
  const onChange = (value: any) => {
    props.onChange(localToUTC(value));
  };

  return (
    <TextWidget
      {...props}
      value={value}
      type="datetime-local"
      onChange={onChange}
    />
  );
};

export default DateTimeWidget;
