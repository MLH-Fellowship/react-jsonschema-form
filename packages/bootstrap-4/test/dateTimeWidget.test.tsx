import React from "react";
import DateTimeWidget from "../src/DateTimeWidget";
import renderer from "react-test-renderer";
import { makeWidgetMockProps } from "./helpers/createMocks";

describe("DateTimeWidget", () => {
  test("simple", () => {
    const tree = renderer
      .create(<DateTimeWidget {...makeWidgetMockProps({})} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
