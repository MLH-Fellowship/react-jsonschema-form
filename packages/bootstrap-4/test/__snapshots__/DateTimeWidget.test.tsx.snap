// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimeWidget simple 1`] = `
<div
  className="mb-0 form-group"
>
  <label
    className="text-danger form-label"
    htmlFor="_id"
  >
    Some simple label
    *
  </label>
  <input
    autoFocus={true}
    className="is-invalid form-control"
    disabled={true}
    id="_id"
    multiple={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={true}
    type="datetime-local"
    uiSchema={Object {}}
    value="2020-07-22T14:42:08.208"
  />
</div>
`;
